app/processImage.js

```diff
@@ -46,6 +48,26 @@ function getCropFunction(cropType) {
     return functionMapping[cropType.toLowerCase()]
 }
 
+function validateImageCropSize(image, size, callback) {
+    if (size.height && size.width) {
+        callback(null, image, size);
+    } else {
+        sharp(image)
+            .metadata()
+            .then(function(metadata) {
+                const asp_ratio = metadata.width/metadata.height;
+                if (!size.height) {
+                    size.height = Math.round(size.width/asp_ratio);
+                }
+                if (!size.width) {
+                    size.width = Math.round(size.height * asp_ratio);
+                }
+            })
```

Comment:
Hi @harshal27, There seems some misunderstanding w.r.t comment https://github.com/PrimeDigitalGlobal/image-resize-lambda/pull/12/files/ae7f762b528f745f673a3a964fbe84a8632fa10c#r158953153 

So basically for handling Javascript or Nodejs promises we use then which takes 2 callback functions, One which handles success state and 2nd one which handles error state.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then  

The above code still doesn't handle the error state. The error state will occur in cases like the image variable contains a file that is not actually a image but something else say a word document. In that case node will not return the error and that request will be stuck in node async loop. and will never return the response back to the client. 

We should always handle these cases in nodejs. The above code can be modified as 

```sharp(image)
             .metadata()
             .then(function(metadata) {
                 const asp_ratio = metadata.width/metadata.height;
                 if (!size.height) {
                     size.height = Math.round(size.width/asp_ratio);
                 }
                 if (!size.width) {
                     size.width = Math.round(size.height * asp_ratio);
                };  
               callback(null, image, size); // Success callback, Image and size will be forwarded to next waterfall step
             }, callback);  // this 2nd callback is the error callback, so that sharp lib here will send the 1st param as error and the control will go to https://github.com/harshal27/image-resize-lambda/blob/24bcc25b8bc97f250c634d3b7672b7387a5ff4be/app/processImage.js#L41 which will handle response for the server/lambda.
```